Add timestamp to logging

chillipeper opened the pull request:
# Description

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

I found the need to add the timestamps to opsdroid logging. This PR is to add the ability to add timestamp to the logging module, hopefully will be useful for others

## Status
**READY** 


## Type of change

<!-- Please delete options that are not relevant. -->


- New feature (non-breaking change which adds functionality)
- Documentation (fix or adds documentation)


# How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. -->

- Tests added to testsuite



# Checklist:

- [X ] I have performed a self-review of my own code
- [ X] I have made corresponding changes to the documentation (if applicable)
- [ X] I have added tests that prove my fix is effective or that my feature works
- [ X] New and existing unit tests pass locally with my changes

jacobtomlinson commented on opsdroid/logging.py:
```diff
@@ -84,12 +84,15 @@ def configure_logging(config):
-            )
+            formatter = logging.Formatter("%(levelname)s %(name)s.%(funcName)s(): %(message)s")
+            if config["logging"]["timestamp"]:
+                formatter = logging.Formatter("%(asctime)s %(levelname)s %(name)s: %(message)s")
```
This has all become a bit messy. It would be nice to build our format string in pieces and then create one `logging.Formatter` object.

Could I ask you to refactor this?

chillipeper replied:
for sure, I was thinking the same, but wasn't sure if it would be your preference. will work on something

jacobtomlinson commented on opsdroid/tests/test_logging.py:
```diff
@@ -123,6 +123,16 @@ def test_configure_console_whitelist():
+    rootlogger = logging.getLogger()
+    assert len(rootlogger.handlers) == 1
+    assert isinstance(rootlogger.handlers[0], logging.StreamHandler)
+
```
It would be good to test that log messages actually have the timestamp in.

jacobtomlinson reviewed (commented):
This generally looks great. However, I think the code needs refactoring a little.

Also the config schema will need updating with the new option.

https://github.com/opsdroid/opsdroid/blob/480cbaa2467597f6e12b883efad1304b447eb763/opsdroid/configuration/validation.py#L8-L13

Cadair commented:
Drive by comment from just the title: please make this configurable!!

chillipeper commented:
I have added the changes requested. As far of testing I have added a regex, as timestamp will be different each time the test is executed. If you have other suggestions let me know

jacobtomlinson reviewed (approved):
Thanks so much @chillipeper! I really appreciate you responding to the review feedback so quickly.
